### PR TITLE
Merged the query input into msg

### PR DESCRIPTION
**Pull request by itwbaer**
The msg input function now executes the query code before sending a
message if that command is given. This opens the channel and then sends
the message, addressing issue #3009.

### CODE REVIEW

File: src/plugins/inputs/msg.js [hunk 1]
```diff
@@ -1,9 +1,63 @@
...
+				type: Msg.Type.ERROR,
+				text: "You cannot open a query window without an argument.",
+			}));
+			return;
+		}
+
+		const query = _.find(network.channels, {name: target});
+
+		if (typeof query !== "undefined") {
+			return;
```
Comment (xPaw) [inline src/plugins/inputs/msg.js]: This return means if there's a query open already, the message will not be sent.

File: src/plugins/inputs/msg.js [hunk 2]
```diff
@@ -1,9 +1,63 @@
...
+const Msg = require("../../models/msg");
+
+exports.commands = ["msg", "say", "query"];
 
 exports.input = function(network, chan, cmd, args) {
-	const target = cmd === "msg" ? args.shift() : chan.name;
+	const target = cmd === "say" ? chan.name : args.shift();
+
+	if (cmd === "query") {
+		if (args.length === 0 || target.length === 0) {
```
Comment (xPaw) [inline src/plugins/inputs/msg.js]: I think due to the args.shift above, this will break opening query windows without any message. `/query user`.

Comment (itwbaer) [reply src/plugins/inputs/msg.js]: You were correct, this broke direct message.

**Comment by itwbaer** [comment]
I tried to squash my merges and screwed my branch up (sorry I am new to this). Just going to close it out.

**Comment by infosisio** [comment]
@itwbaer Is there a particular barrier pending?

**Pull request closed without merging**